src/basic/math/bc44.convert_i16_to_FAC1.s

```diff
@@ -0,0 +1,29 @@
+;; #LAYOUT# STD *       #TAKE
+;; #LAYOUT# X16 BASIC_0 #TAKE-OFFSET 2000
+;; #LAYOUT# *   BASIC_0 #TAKE
+;; #LAYOUT# *   *       #IGNORE
+
+;
+; Math package - Convert signed 16-bit integer to floating point
+;
+; - https://codebase64.org/doku.php?id=base:kernal_floating_point_mathematics
+; - https://www.c64-wiki.com/wiki/GIVAYF (which calls this function)
+;
+; Convert the signed 16-bit integer in FAC1 mantissa ($62:$63)
+; into a floating point number in FAC1
+
+; NOTE:
+; According to https://c64os.com/post/floatingpointmath at $BC49
+; Assume FAC1_mantissa+0 and +1. X=exponent and C=0 if negative and C=1 if positive
+
+convert_i16_to_FAC1:
+    ldx #$90
+    jsr carry
+
+    jmp convert_i16_expsign_to_FAC1   ; This should be at $BC49 (see note above)
+
+
+carry:
```

Comment:
I have a question: why is this a subroutine? Is this intended to be called from somewhere else? JSR + address + RTS are 4 bytes total.

Reply:
At $BC49 according to https://c64os.com/post/floatingpointmath there should be a function having the exponent in the x register, the carry flag containing the sign and FAC1 having the 16-bit integer (I think still signed). Since `convert_i16_to_FAC1` should be at $BC44 we only have 5 bytes to set x to the exponent and transfer the sign to the carry flag. The ldx takes 2 bytes and then there is only room left for a `jsr` to transfer the sign. After this I first put the rest of the code to do the conversion in this file, but there isn't enough room so I created the separate `convert_i16_expsign_to_FAC1`. I hope this makes sense.

Reply:
Right, you even put a comment. at the top of the file. OK, I think we can merge it.

Reply:
BTW, it could be actually be worthwhile to try to enforce the `convert_i16_to_FAC1` to start from $BC49 and move the `carry` function to a separate `.s` file. We would save 3 bytes on the `jmp` - but it's sometimes hard to do due to address conflicts.